src/CodeGen_ARM.cpp

```diff
@@ -1261,4 +1261,14 @@ int CodeGen_ARM::native_vector_bits() const {
     return 128;
 }
 
+bool CodeGen_ARM::target_needs_software_cast_from_float16_to(Type t) const {
```

Comment:
Are we ever going to encounter a target with hardware support for casts from float16 but not hardware support for casts to float16?

I suggest simplifying this pair of methods into something like "has_hardware_float16_cast"